File: Food.cpp
#include "Food.hpp"
#include "Aquarium.hpp"

Food::Food(Aquarium& _aquarium, float x) : AquariumObject(_aquarium, 2, Vector2(x, 0)) {
    
}

bool Food::operator==(const Food& other) const {
    return this == &other;
}

void Food::move() {
    setPosition(getPosition() + Vector2::down * getSpeed());
    if (getPosition().y >= getAquarium().getSizeY()) {
        destruct();
    }
}

void Food::destruct() {
    getAquarium().remove(*this);
}

File: production/Java/src/Food.java
public class Food extends AquariumObject implements IDestructible {
  public static final int price = 5;

  public Food(Aquarium aquarium, double abscissa) {
    super(aquarium, 2, new Vector2(abscissa, 0));
  }

  public void move() {
    setPosition(getPosition().add(Vector2.down.multiply(getSpeed())));
  }

  public void tick() {
    move();
    if (getPosition().ordinate >= getAquarium().getSizeY()) {
      destruct();
    }
  }

  public void destruct() {
    getAquarium().remove(this);
  }
}